dashboard/lib/client_state.rb

```diff
@@ -118,30 +117,22 @@ def add_callout_seen(callout_key)
 
   private
 
-  # Migrates cookies[:progress] back to session[:progress] in case we end up rolling
-  # back the release that does the forward migration. Except in that case, this code
-  # does nothing becaue we don't otherwise use cookies[:progress] and cookies[:lines].
-  def undo_cookie_migration_on_rollback
-    if cookies[:progress].present? && session[:progress].blank?
+  def progress_hash
```

Comment:
did you know that in ruby you can rescue an entire method without a begin/end block?

````
  def progress_hash
    migrate_cookies
    if cookies[:progress]
      JSON.parse(cookies[:progress])
    else
      {}
    end
    rescue JSON::ParserError
      {}
  end
````

(your choice whether you like the ruby-ism of not using return when not necessary for control flow...)